pkg/pool-weighted/contracts/WeightedPool.sol

```diff
@@ -16,11 +16,12 @@ pragma solidity ^0.7.0;
 pragma experimental ABIEncoderV2;
 
 import "./BaseWeightedPool.sol";
+import "./InvariantGrowthProtocolFees.sol";
 
 /**
  * @dev Basic Weighted Pool with immutable weights.
  */
-contract WeightedPool is BaseWeightedPool {
+contract WeightedPool is BaseWeightedPool, InvariantGrowthProtocolFees {
```

Comment:
I think the proper way would be to avoid inheriting from `BaseWeightedPool` in `InvariantGrowthProtocolFees` while denoting here we do want to extend `BaseWeightedPool` using `InvariantGrowthProtocolFees`'s methods. Although, I think the only way to do this would be using a common interface between `BaseWeightedPool` and `InvariantGrowthProtocolFees`

Reply:
Agreed, though I think the current mechanism is fine for now. I wouldn't stress over this until we have more variants.